src/core/utils/fs.cpp

```diff
@@ -516,3 +521,61 @@ QString Utils::Fs::cacheLocation()
         locationDir.mkpath(locationDir.absolutePath());
     return location;
 }
+
+#ifndef DISABLE_GUI
+// Open the given path with an appropriate application
+void Utils::Fs::openPath(const QString& absolutePath)
+{
+    const QString path = fromNativePath(absolutePath);
+    // Hack to access samba shares with QDesktopServices::openUrl
+    if (path.startsWith("//"))
+        QDesktopServices::openUrl(toNativePath("file:" + path));
+    else
+        QDesktopServices::openUrl(QUrl::fromLocalFile(path));
+}
+
+// Open the parent directory of the given path with a file manager and select
+// (if possible) the item at the given path
+void Utils::Fs::openFolderSelect(const QString& absolutePath)
+{
+    const QString path = fromNativePath(absolutePath);
+#ifdef Q_OS_WIN
+    if (QFileInfo(path).exists()) {
+        // Syntax is: explorer /select, "C:\Folder1\Folder2\file_to_select"
+        // Dir separators MUST be win-style slashes
+        QProcess::startDetached("explorer.exe /select,\"" + toNativePath(path) + "\"");
+    }
+    else {
+        // If the item to select doesn't exist, try to open its parent
+        openPath(path.left(path.lastIndexOf("/")));
+    }
+#elif defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
+    if (QFileInfo(path).exists()) {
+        QProcess proc;
+        QString output;
+        proc.start("xdg-mime", QStringList() << "query" << "default" << "inode/directory");
+        proc.waitForFinished();
+        output = proc.readLine().simplified();
+        if (output == "dolphin.desktop")
+            proc.startDetached("dolphin", QStringList() << "--select" << toNativePath(path));
+        else if (output == "nautilus.desktop" || output == "org.gnome.Nautilus.desktop"
+                 || output == "nautilus-folder-handler.desktop")
+            proc.startDetached("nautilus", QStringList() << "--no-desktop" << toNativePath(path));
+        else if (output == "caja-folder-handler.desktop")
+            proc.startDetached("caja", QStringList() << "--no-desktop" << toNativePath(path));
+        else if (output == "nemo.desktop")
+            proc.startDetached("nemo", QStringList() << "--no-desktop" << toNativePath(path));
+        else if (output == "kfmclient_dir.desktop")
+            proc.startDetached("konqueror", QStringList() << "--select" << toNativePath(path));
+        else
+            openPath(path.left(path.lastIndexOf("/")));
```

Comment:
`path` is the path to the file, if you do `openPath(path)` the file will be opened with a proper application. We want to start a filemanager, so here I'm removing the name of the file so that it's parent is opened.